src/code/code_80069420.c

```diff
@@ -1,23 +1,48 @@
 #include "global.h"
 
-void* MemCopy(void* dest, void* src, s32 size) {
-    u8* destu = (u8*)dest;
-    u8* srcu = (u8*)src;
+/**
+ * @brief memcpy: copies @p len bytes from memory starting at @p src to memory starting at @p dest . Expects the memory
+ * specified by @p src and @p dest to not overlap.
+ *
+ * @sa libultra also has a memcpy().
+ *
+ * @param[in,out] dest address of start of buffer writing to
+ * @param[in] src address of start of buffer to read from
+ * @param[in] len number of bytes to copy. Has to be s32: it is not possible to copy more than 2 GB of RAM at once.
+ *
+ * @return dest
+ */
+void* MemCpy(void* dest, const void* src, s32 len) {
+    u8* d = dest;
+    const u8* s = src;
 
-    while (size > 0) {
-        *destu++ = *srcu++;
-        size--;
+    while (len > 0) {
+        *d++ = *s++;
+        len--;
     }
 
     return dest;
 }
 
-void* MemSet(void* dest, s32 val, s32 size) {
-    u8* destu = (u8*)dest;
-    s32 s = size;
+/**
+ * @brief memset: sets @p len bytes to @p val starting at address @p dest .
+ *
+ * There are two other memsets in this codebase,
+ * @sa Lib_MemSet(), __osMemSet()
+ * This one is unused.
+ *
+ * @param[in,out] dest address to start at
+ * @param[in] val value to write (s32, but interpreted as u8)
+ * @param[in] len number of bytes to write. Has to be s32: it is not possible to set more than 2 GB of RAM at once.
```

Comment:
Mm, I suppose it would probably be better to note that this cannot sensibly be a `size_t` (The function argument can, but the temp must be `s32`, and it feels disingenuous to allow the user to accidentally feed it a large integer that gets recast into a small negative one.